cartridge/test-helpers/cluster.lua

```diff
@@ -156,6 +156,39 @@ function Cluster:server(alias)
     error('Server ' .. alias .. ' not found', 2)
 end
 
+--- Return iterator for cluster server's with enabled role
+local function iter_servers_by_role(cluster, role_name)
+    local replicasets_with_role = fun.iter(cluster.replicasets)
+    :filter(function(rs)
+        return fun.iter(rs.roles)
+            :filter(function(role)
+                return role == role_name
+            end)
+            :nth(1)
+    end)
+    :map(function(rs) return rs.uuid, true end)
+    :tomap()
+
+    return fun.iter(cluster.servers)
+        :filter(function(server)
+            return replicasets_with_role[server.replicaset_uuid]
+        end)
+end
```

Comment:
To be honest patch looks like you first time faced luafun and tried to use it everywhere.

It could be easily rewritten as
```lua
local function iter_servers_by_role(cluster, role_name)
    local replicasets_with_role = fun.iter(cluster.replicasets)
        :map(function(rs) return rs.uuid, rs.roles end)
        :map(function(rs_uuid, roles) return rs_uuid, utils.table_find(roles, role_name) ~= nil end)
        :tomap()

    return fun.iter(cluster.servers)
        :filter(function(server)
            return replicasets_with_role[server.replicaset_uuid]
        end)
end
```

For me it's more clear and laconic.

Reply:
Also I hope you won't write production code in such way. Huge amount of closures is a perfect performance killer.

Reply:
> you first time faced luafun and tried to use it everywhere

`luafun` is not the only tool that allows you to use FP :)
But you'r right, its first time.

> Also I hope you won't write production code in such way. Huge amount of closures is a perfect performance killer.

Thanks for your advice. I like your version a little more, but this does not affect the execution speed in any way.
For test it i increased count of replicasets in helper cluster by 1k.
PS
> Huge amount of closures is a perfect performance killer

I agree with that,  but you have the same number of closures as me, dont you?

Reply:
> I agree with that, but you have the same number of closures as me, dont you?

It's not so critical for tests. I told about production code.

Reply:
@palage4a would you apply that refactoring?

Reply:
> @palage4a would you apply that refactoring?

yes, sorry for delay. I will try update PR today.